py4cast/datasets/poesy/__init__.py

```diff
@@ -112,22 +116,40 @@ def load_data(
     return data_array[
         param.grid.subdomain[0] : param.grid.subdomain[1],
         param.grid.subdomain[2] : param.grid.subdomain[3],
-        term,
+        (term / dt.timedelta(hours=1)).astype(int) - 1,
         member,
     ]
 
 
-def exists(ds_name: str, param: WeatherParam, date: dt.datetime) -> bool:
-    flist = get_filepath(ds_name, param, date)
-    return flist.exists()
+def exists(
+    ds_name: str,
+    param: WeatherParam,
+    timestamps: Timestamps,
+    file_format: Literal["npy", "grib"] = "grib",
+) -> bool:
+
+    filepath = get_filepath(ds_name, param, timestamps.datetime, file_format)
+    if not filepath.exists():
+        return False
+    return True
+
+
+def valid_timestamp(n_inputs: int, timestamps: Timestamps):
```

Comment:
Consider adding docstrings for the 2 functions above, exists and valid_timestamp, I propose:
"""
Verification function called after the creation of each timestamps, check if computed terms respect the dataset convention.
Reminder:
Poesy terms are between 0 and 44.
"""

"""
Return True if the file exists on the disk.
"""